pkg/syncer/syncer.go

```diff
@@ -67,32 +68,48 @@ func (c *Controller) handleErr(err error, i interface{}) {
 	// Re-enqueue up to 5 times.
 	num := c.Queue.NumRequeues(i)
 	if num < 5 {
-		log.Printf("Error reconciling key %q, retrying... (#%d): %v", i, num, err)
+		klog.Errorf("Error reconciling key %q, retrying... (#%d): %v", i, num, err)
 		c.Queue.AddRateLimited(i)
 		return
 	}
 
 	// Give up and report error elsewhere.
 	c.Queue.Forget(i)
 	utilruntime.HandleError(err)
-	log.Printf("Dropping key %q after failed retries: %v", i, err)
+	klog.Errorf("Dropping key %q after failed retries: %v", i, err)
 }
 
 func (c *Controller) process(gvr schema.GroupVersionResource, obj interface{}) error {
+	klog.Infof("Process object of type: %T : %v", obj, obj)
 	obj, exists, err := c.FromDSIF.ForResource(gvr).Informer().GetIndexer().Get(obj)
 	if err != nil {
+		klog.Error(err)
 		return err
 	}
+	/*
```

Comment:
Is `interfaceToUnstructured` wrong in some way? If so we should delete it. If not we should at least have a TODO about how to fix it here.

Reply:
It is not necessary since the object is already of this type (at least in the case of CRDs). And before removing it I was having an error ("Kind should not be empty"), occurring during the conversion.

I'll remove the function then.